lib/Onyx.js

```diff
@@ -1049,7 +1068,7 @@ function set(key, value) {
     const updatePromise = broadcastUpdate(key, valueWithNullRemoved, hasChanged, 'set');
 
     // If the value has not changed, calling Storage.setItem() would be redundant and a waste of performance, so return early instead.
-    if (!hasChanged) {
+    if (!hasChanged || ongoingClear) {
```

Comment:
Looks like the second operand never works here, because of a condition above, where you already checked `ongoingClear` equals `true`.

Reply:
This one can be removed